### PR TITLE
Stops ninja nets from removing plasmaman suits+internals

### LINKED ISSUE #37517: Plasmamen spawn with an oxy tank in their box
You know the box everyone spawns with, that has a small oxytank, breath mask etc?

Plasmamen also receive an oxygen tank inside their box, which is pretty bad for a race that needs to breathe plasma so they don‘t die.

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
fix: The Spider Clan has issued an apology for stripping plasmamen of their protective suits and internals, stating that it was "kind of a dick move" and "will never happen again".
tweak: The Spider Clan holding facility has completed minor renovations. The dojo now includes medical and surgical equipment, while the lobby has a fire extinguisher and plasmaman internals. 
/:cl:

Fixes: #37517

I essentially replaced the net TP code with the gulag teleporter's and added plasmaman internals as well as a bunch of fluff to the holding facility.

Also:
Added a plasma internals locker since the TP will still remove plasma tanks if they're stored inside the suit slot (and the suit gets left behind).

Map edits unrelated to plasmamen:
The dojo now has medical items (defibs, synthflesh, surgical equipment, blood bags) so that people can actually fight more than once. Bar table has been replaced with the regular type that doesn't throw players.
Kitchen has a soap bar, box of spare lights and carp meat, hydroponics area has a spade/cultivator/bucket, restroom has a urinal, lathe area has a multitool.

### CODE REVIEW

File: code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm [hunk 1]
```diff
@@ -9,20 +9,26 @@ It is possible to destroy the net by the occupant or someone else.
...
-	mouse_opacity = MOUSE_OPACITY_ICON//So you can hit it with stuff.
-	anchored = TRUE//Can't drag/grab the net.
+	density = TRUE //Can't pass through.
+	opacity = 0 //Can see through.
+	mouse_opacity = MOUSE_OPACITY_ICON //So you can hit it with stuff.
+	anchored = TRUE //Can't drag/grab the net.
 	layer = ABOVE_ALL_MOB_LAYER
 	max_integrity = 25 //How much health it has.
 	can_buckle = 1
 	buckle_lying = 0
```
Comment (ShizCalev) [inline code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm]: false

File: code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm [hunk 2]
```diff
@@ -9,20 +9,26 @@ It is possible to destroy the net by the occupant or someone else.
...
-	opacity = 0//Can see through.
-	mouse_opacity = MOUSE_OPACITY_ICON//So you can hit it with stuff.
-	anchored = TRUE//Can't drag/grab the net.
+	density = TRUE //Can't pass through.
+	opacity = 0 //Can see through.
+	mouse_opacity = MOUSE_OPACITY_ICON //So you can hit it with stuff.
+	anchored = TRUE //Can't drag/grab the net.
 	layer = ABOVE_ALL_MOB_LAYER
 	max_integrity = 25 //How much health it has.
 	can_buckle = 1
```
Comment (ShizCalev) [inline code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm]: true

File: code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm [hunk 3]
```diff
@@ -9,20 +9,26 @@ It is possible to destroy the net by the occupant or someone else.
 	icon = 'icons/effects/effects.dmi'
 	icon_state = "energynet"
 
-	density = TRUE//Can't pass through.
-	opacity = 0//Can see through.
-	mouse_opacity = MOUSE_OPACITY_ICON//So you can hit it with stuff.
-	anchored = TRUE//Can't drag/grab the net.
+	density = TRUE //Can't pass through.
+	opacity = 0 //Can see through.
```
Comment (ShizCalev) [inline code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm]: false

File: code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm [hunk 4]
```diff
@@ -66,7 +73,7 @@ It is possible to destroy the net by the occupant or someone else.
 	affecting.forceMove(pick(GLOB.holdingfacility)) //Throw mob in to the holding facility.
 	to_chat(affecting, "<span class='danger'>You appear in a strange place!</span>")
 
-	if(!QDELETED(master))//As long as they still exist.
+	if(!QDELETED(master)) //As long as they still exist.
 		to_chat(master, "<span class='notice'><b>SUCCESS</b>: transport procedure of [affecting] complete.</span>")
 	do_sparks(5, FALSE, affecting)
 	playsound(affecting, 'sound/effects/phasein.ogg', 25, 1)
```
Comment (ShizCalev) [inline code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm]: TRUE

File: code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm [hunk 5]
```diff
@@ -53,11 +59,12 @@ It is possible to destroy the net by the occupant or someone else.
...
-				continue
-			H.dropItemToGround(W)
+			if(!is_type_in_typecache(W, weeb_jail_essential_items) && H.temporarilyRemoveItemFromInventory(W)) //Remove all non-essential items, including implants.
+				if(istype(W, /obj/item/restraints/)) //No need for restraints once they're inside the weeb jail.
+					W.forceMove(get_turf(H.loc))
+					continue
+				else
+					W.forceMove(get_turf(H.loc))
 
 	playsound(affecting, 'sound/effects/sparks4.ogg', 50, 1)
```
Comment (ShizCalev) [inline code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm]: TRUE

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Arianya** [comment]
Aren't the wizard medbots sprayed liberally throughout the area meant to help people not die from arena fighting etc?

**Comment by 81Denton** [comment]
I moved the arena bot outside the cage so it won't inject people and get in the way mid fight. The other one is in the same spot as before, just stationary so it won't follow people outside the spawn area.

File: code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm (hunk 3 above)
Comment (81Denton) [reply code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm]: http://www.byond.com/docs/ref/info.html#/atom/var/opacity
Name is a bit misleading

**Comment by 81Denton** [comment]
Just noticed that the hydroponics crate was empty, hydro job guide/multitool/spare lights/soap for microwave/urinal were missing.

**Comment by GuppyLaxx** [comment]
just let me DIE

File: code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm [hunk 6]
```diff
@@ -53,31 +59,32 @@ It is possible to destroy the net by the occupant or someone else.
...
-				continue//So all they're left with are shoes and uniform.
-			if(W == H.shoes)
-				continue
-			H.dropItemToGround(W)
+			if(!is_type_in_typecache(W, weeb_jail_essential_items) && H.temporarilyRemoveItemFromInventory(W)) //Remove all non-essential items, including implants.
+				if(istype(W, /obj/item/restraints/)) //No need for restraints once they're inside the weeb jail.
+					W.forceMove(get_turf(H.loc))
+					continue
+				else
+					W.forceMove(get_turf(H.loc))
```
Comment (ShizCalev) [inline code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm]: What's the point of this else?

Comment (81Denton) [reply code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm]: It's a leftover from gulag TP code, I'll remove it when I'm back from work.

File: code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm [hunk 7]
```diff
@@ -53,31 +59,30 @@ It is possible to destroy the net by the occupant or someone else.
...
 		var/mob/living/carbon/human/H = affecting
 		for(var/obj/item/W in H)
-			if(W == H.w_uniform)
-				continue//So all they're left with are shoes and uniform.
-			if(W == H.shoes)
-				continue
-			H.dropItemToGround(W)
+			if(!is_type_in_typecache(W, weeb_jail_essential_items) && H.temporarilyRemoveItemFromInventory(W)) //Remove all non-essential items, including implants.
+				if(istype(W, /obj/item/restraints/)) //No need for restraints once they're inside the weeb jail.
+					W.forceMove(get_turf(H.loc))
```
Comment (ShizCalev) [inline code/modules/ninja/suit/n_suit_verbs/energy_net_nets.dm]: h.drop_location()?

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by 81Denton** [comment]
I won't be able to work on this for a bit, so I'll close it for now.

**Pull request closed without merging**